middleman-core/lib/middleman-core/preview_server.rb

```diff
@@ -110,24 +111,29 @@ def new_app
           run meta_app
         end
 
+        my_ip       = opts[:bind_address].presence || public_ip
+        my_hostname = opts[:host].presence || resolve_ip(my_ip) || Socket.gethostname.tr(' ', '+')
```

Comment:
The problem is this, `middleman` tells you it's listening on `http://host.in.mydomain.org:4567/`

```
% middleman
== Could not find a Middleman project config.rb
== Treating directory as a static site to be served
== The Middleman is loading
== The Middleman is standing watch at http://host.in.mydomain.org:4567/ (http://192.168.1.2:4567/)
== Inspect your site configuration at http://host.in.mydomain.org:4567/__middleman
```

Looking at `netstat` you get this:

```
netstat -nlpt
tcp        0      0 0.0.0.0:4567            0.0.0.0:*               LISTEN      7222/ruby    
```

But doing a reverse lookup in a SOHO-environment with a simple dns forwarder fails for the local ip address:

```
host 192.168.1.2
Host 2.1.168.192.in-addr.arpa. not found: 3(NXDOMAIN)
```

and a normal lookup does not work either

```
% host $(hostname)
Host host.in.mydomain.org not found: 3(NXDOMAIN)
```

> reverse DNS lookup at any point (because they were on the train or whatever)

I think that was me using my coporate account. The resolve_ip thing is only used if you do not pass a hostname. I will investigate a little bit further and will try to build a solution which works for a coporate and for SOHO-environments. Ok?